frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/footer/Footer.component';
+import Header from '~/components/header/Header.component';
+
+import styles from '../styles/pages/default-page.module.scss';
+import style from './submissions.module.scss';
+import ExSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import { loadSubmissions } from '~/redux/exercise-submissions/exercise-submissions.actions';
+import SubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import { Submission } from '~/redux/exercise-submissions/types';
+import SearchExerciseSubmissions from './searchExerciseSubmissions';
+
+interface State {
+  loading: boolean;
+}
+interface Props {
+  submissions: any;
+  dispatch: any;
+}
+
+class Submissions extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+    const state = submissions;
+    if (state.submission === undefined) {
+      this.getData();
+    }
+  }
+
+  getData = async () => {
+    const { dispatch } = this.props;
+    try {
+      const response = await ExerciseService.getSubmissions(0, '');
+      const jsonResp = await response.json();
+      return dispatch(loadSubmissions(jsonResp));
+    } catch (err) {
+      return SweetAlertService.toast({ type: 'error', text: err });
```

Comment:
Some here, a small nitpick but to keep things as minimal as possible, no need for `return`.